src/main/java/org/variantsync/diffdetective/variationtree/VariationNode.java

```diff
@@ -0,0 +1,583 @@
+     */
+    public abstract List<String> getLabelLines();
+
+    /**
+     * Returns the start line number of the {@link getLabelLines label} of this node in the
+     * corresponding source code.
+     *
+     * @see setFromLine
+     */
+    public abstract int getFromLine();
+
+    /**
+     * Changes the line number returned by {@link getFromLine}.
+     *
+     * @see getFromLine
+     */
+    public abstract void setFromLine(int from);
+
+    /**
+     * Returns the end line number of the {@link getLabelLines label} of this node in the
+     * corresponding source code.
+     *
+     * <p>The line number is exclusive (i.e., it points 1 behind the last included line).
+     *
+     * @see setToLine
+     */
+    public abstract int getToLine();
+
+    /**
+     * Changes the line number returned by {@link getToLine}.
+     *
+     * @see getToLine
+     */
+    public abstract void setToLine(int to);
+
+    /**
+     * Returns the range of line numbers of this node's corresponding source code.
+     *
+     * @see getFromLine
+     * @see getToLine
+     */
+    public Lines getLineRange() {
+        return Lines.FromInclToExcl(getFromLine(), getToLine());
+    }
+
+    /**
+     * Returns the parent of this node, or {@code null} if this node doesn't have a parent.
+     *
+     * @see drop
+     * @see addBelow
+     */
+    public abstract VariationNode<T> getParent();
+
+    /**
+     * Returns an unmodifiable list representing the children of this node.
+     *
+     * <p>The following invariant has to hold for all {@code node}s:
+     * <code>
+     *   for (var child : node.getChildren()) {
+     *     Assert.assertTrue(node.isSameAs(child.getParent(node)))
+     *   }
+     * </code>
+     *
+     * @see isChild
+     * @see addChild
+     * @see removeChild
+     */
+    public abstract List<VariationNode<T>> getChildren();
+
+    /**
+     * Returns {@code true} iff this node has no parent.
+     *
+     * @see getParent
+     */
+    public boolean isRoot() {
+        return getParent() == null;
+    }
+
+    /**
+     * Returns {@code true} iff this node has no children.
+     *
+     * @see getChildren
+     */
+    public boolean isLeaf() {
+        return getChildren().isEmpty();
+    }
+
+    /**
+     * Returns the number of child nodes.
+     *
+     * <p>Note: This is O(n) for {@link DiffNode.Projection}.
+     *
+     * @see getChildCount
+     */
+    public int getChildCount() {
+        return getChildren().size();
+    }
+
+    /**
+     * Computes the length of the path from the root node to this node.
+     */
+    public int getDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        return getParent().getDepth() + 1;
+    }
+
+    /**
+     * Computes the number of {@link isAnnotation annotations} of the path from the root node to
+     * this node.
+     */
+    public int getAnnotationDepth() {
+        if (isRoot()) {
+            return 0;
+        }
+
+        if (isIf()) {
```

Comment:
or Elif or Else? Where is this method used?

Reply:
Oh. I think the comment is misleading. I didn't actually write this method so the implementation seems correct (count the number of if-elif-else chains) but I got the comment wrong..

It was previously implemented in `DiffNode` and is currently unused. It was introduced in f72d9552748dcc1fc024c130b6cef391ac016791 (as `getBeforeAnnotationDepth` and `getAfterAnnotationDepth`) and the last usage was removed in 4ac9cdaf3f89675ebd35a1918f52b10fb030a93c (December 2021).

I have no idea if this method will ever be used again. What do you think about removing it?

Reply:
I agree to remove it.